sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -222,18 +231,18 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
   }
 
   private def verifySchema(schema: StructType): Unit = {
-    def verifyType(dataType: DataType): Unit = dataType match {
+    def verifyType(name: String, dataType: DataType): Unit = dataType match {
         case ByteType | ShortType | IntegerType | LongType | FloatType |
              DoubleType | BooleanType | _: DecimalType | TimestampType |
              DateType | StringType =>
 
-        case udt: UserDefinedType[_] => verifyType(udt.sqlType)
+        case udt: UserDefinedType[_] => verifyType(name, udt.sqlType)
 
         case _ =>
           throw new UnsupportedOperationException(
```

Comment:
I am unhappy that CSV currently throws `UnsupportedOperation` but text datasource throws `AnalysisException`. I just matched this to `UnsupportedOperation`; however, I am happy to match this to `AnalysisException` if anyone thinks so.